src/ProgressiveColonizationSystem/PksTierRequirement.cs

```diff
@@ -0,0 +1,75 @@
+﻿using ContractConfigurator;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace ProgressiveColonizationSystem
+{
+    public class PksTierRequirement : ContractRequirement
+    {
+        private string body;
+        private string researchCategory;
+        private int tier;
+
+        public override void OnSave(ConfigNode configNode)
+        {
+            configNode.AddValue(nameof(body), this.body);
+            configNode.AddValue(nameof(researchCategory), this.researchCategory);
+            configNode.AddValue(nameof(tier), this.tier);
+        }
+
+        public override void OnLoad(ConfigNode configNode)
+        {
+            this.LoadFromConfig(configNode);
+        }
+
+        protected override string RequirementText()
+        {
+            return $"Must have reached tier-{this.tier} {this.researchCategory} on {this.body}";
+        }
+
+        public override bool LoadFromConfig(ConfigNode configNode)
```

Comment:
So I believe I got this right, but yet, there are warning messages coming from this about "unknown config value, 'body'..."